lib/BlockedBy.js

```diff
@@ -103,6 +103,41 @@ async function blockedBySelf({ waitingKey, buildId, collapse }) {
 
     return false;
 }
+
+/**
+ *
+ * @param {String} buildConfig
+ * @param {Array} blockingBuildIds
+ * @param {String} buildId
+ * @param {Object} redisConn
+ */
+async function checkMultipleBuildsFromSameEvent(buildConfig, blockingBuildIds, buildId, redisConn) {
+    if (!buildConfig) return false;
+
+    const parsedConfig1 = JSON.parse(buildConfig);
+
+    if (!parsedConfig1 || !parsedConfig1.eventId) {
+        return false;
+    }
+
+    const isSameEvent = await Promise.race(blockingBuildIds.map(async (b) => {
+        const bConfigs = await redisConn.hget(
```

Comment:
does this return build or multipe build ? From subsequent logic it seems like singular, so please call this `blockedBuildJson` (or just call this `blockedBuild` and do `JSON.parse(await ...)`)